templates/python/custom-copilot-rag-azure-ai-search/README.md.tpl

```diff
@@ -0,0 +1,142 @@
+> - A [Microsoft 365 account for development](https://docs.microsoft.com/microsoftteams/platform/toolkit/accounts).
+{{/enableTestToolByDefault}}
+{{#enableTestToolByDefault}}
+> - [Node.js](https://nodejs.org/) (supported versions: 16, 18) for local debug in Test Tool.
+{{/enableTestToolByDefault}}
+
+### Configurations
+1. First, Open the command box and enter `Python: Create Environment` to create and activate your desired virtual environment. Remember to select `requirements.txt` as dependencies to install when creating the virtual environment.
+{{#enableTestToolByDefault}}
+{{#useAzureOpenAI}}
+1. In file *env/.env.testtool.user*, fill in your Azure OpenAI key `SECRET_AZURE_OPENAI_API_KEY`, deployment name `AZURE_OPENAI_MODEL_DEPLOYMENT_NAME`, endpoint `AZURE_OPENAI_ENDPOINT` and embedding deployment name `AZURE_OPENAI_EMBEDDING_DEPLOYMENT`.
+{{/useAzureOpenAI}}
+{{#useOpenAI}}
+1. In file *env/.env.testtool.user*, fill in your OpenAI key `SECRET_OPENAI_API_KEY`. 
+1. In this template, default model name is `gpt-3.5-turbo` and default embedding model name is `text-embedding-ada-002`. If you want to use different models from OpenAI, fill in your model names in [src/config.py](./src/config.py).
+{{/useOpenAI}}
+1. In file *env/.env.local.user*, fill in your Azure Search key `SECRET_AZURE_SEARCH_KEY` and endpoint `AZURE_SEARCH_ENDPOINT`.
+{{/enableTestToolByDefault}}
+{{^enableTestToolByDefault}}
+{{#useAzureOpenAI}}
+1. In file *env/.env.local.user*, fill in your Azure OpenAI key `SECRET_AZURE_OPENAI_API_KEY`, deployment name `AZURE_OPENAI_MODEL_DEPLOYMENT_NAME`, endpoint `AZURE_OPENAI_ENDPOINT` and embedding deployment name `AZURE_OPENAI_EMBEDDING_DEPLOYMENT`.
+{{/useAzureOpenAI}}
+{{#useOpenAI}}
+1. In file *env/.env.local.user*, fill in your OpenAI key `SECRET_OPENAI_API_KEY`. 
+1. In this template, default model name is `gpt-3.5-turbo` and default embedding model name is `text-embedding-ada-002`. If you want to use different models from OpenAI, fill in your model names in [src/config.py](./src/config.py).
+{{/useOpenAI}}
+1. In file *env/.env.local.user*, fill in your Azure Search key `SECRET_AZURE_SEARCH_KEY` and endpoint `AZURE_SEARCH_ENDPOINT`.
+{{/enableTestToolByDefault}}
+
+### Setting up index and documents
+{{^enableTestToolByDefault}}
+1. Azure Search key `SECRET_AZURE_SEARCH_KEY` and endpoint `AZURE_SEARCH_ENDPOINT` are loaded from *env/.env.local.user*. Please make sure you have already configured them.
+{{/enableTestToolByDefault}}
+{{#enableTestToolByDefault}}
+1. Azure Search key `SECRET_AZURE_SEARCH_KEY` and endpoint `AZURE_SEARCH_ENDPOINT` are loaded from *env/.env.testtool.user*. Please make sure you have already configured them.
+{{/enableTestToolByDefault}}
+1. Use command `python -m setup` to create index and upload documents in `src/files`.
+1. You will see the following information indicated the success of setup:
+    ```
+    Create index succeeded. If it does not exist, wait for 5 seconds...
```

Comment:
What does `If it does not exist, wait for 5 seconds...` and `If they do not exist, wait for several seconds...` mean? Do we need to show them to users?

Reply:
it means "if it does not exist in portal". words need to be polished

Reply:
@MuyangAmigo Please help review the log out information